### ISSUE TITLE
[BUG] Date/time with 'enableTime=false' should save value in the submission data in the [YYYY]-[MM]-[DD] ISO-8601 format

### ISSUE THREAD

**Issue by andreinitescu** [formio/formio.js#4178]
## Environment

The issue is reproduceable easily on https://formio.github.io/formio.js/app/builder

## Steps to Reproduce
1.  Add a **Date/Time** component
2.  In the component settings, on the 'Time' tab, uncheck 'Enable Time Input' so the component works as a **Date** selector component
3.  In the generated form, select a date and observe the value in the generated submission data

## Expected behavior

Since the component is configured as a **Date only** component, the value in the submission data **should not** include the **time** and **time-zone**, it should be in the  **[YYYY]-[MM]-[DD]** ISO-8601 format:

```
{
    "data": {
        "submit": false,
        "birthDate": "2021-07-23"
    },
    "metadata": {}
}
```

## Observed behavior

Although the component is configured as a **Date only** component, the value in the submission data **includes** the **time** and **time-zone**, it uses the **[YYYY]-[MM]-[DD]T[hh]:[mm]:[ss]±[hh]:[mm]** ISO-8601 format:

```
{
    "data": {
        "submit": false,
        "birthDate": "2021-07-23T00:00:00+03:00"
    },
    "metadata": {}
}
```

In other words, in a form with a **Date/time component** with **enableTime=false**, which is meant to fill a **date** (for example, a birth date, an expiration date, etc.) it makes no sense to get the submission value with **time** and **time-zone**:
```
"birthDate": "2021-07-23T00:00:00+03:00"
```

More importantly, including the time and time-zone for a Date only component creates problems in technologies where there's a clear difference between a 'date' and a 'date-time', and there's a type for each. Strict validation of the value (not referring to the validation in Form.Io) becomes a problem. 

In the case you think that by changing the value format it could be a backward compatibility problem, I suggest adding a new setting on the component, something like 'Save date only value'.

**Comment by mihaiandreicut** [comment]
I have the same situation, and I found this solution: on 'Data' tab, in Flatpicker option section, add this
`{
  "dateFormat": "yyyy-MM-dd"
}`

**Comment by andreinitescu** [comment]
@mihaiandreicut multumesc :) o sa ma uit

**Comment by andreinitescu** [comment]
@mihaiandreicut Multumesc, that's phenomenal! 👍 Thanks a lot!

Do you know by any chance https://github.com/formio/formio.js/issues/4268
